realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -665,6 +665,9 @@ public static String getTableName() {
     }
 
     public static long insert(Realm realm, some.test.AllTypes object, Map<RealmModel,Long> cache) {
+        if (object instanceof RealmObjectProxy && ((RealmObjectProxy)object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy)object).realmGet$proxyState().getRealm$realm().getPath().equals(realm.getPath())) {
```

Comment:
Minor, but this could be rewritten to something like the below better readability:
```
if (object instanceof RealmObjectProxy) {
  BaseRealm proxyState = ((RealmObjectProxy) object).realmGet$proxyState();
  if (proxyState.getRealm$realm() != null && proxyState.getRealm$realm().getPath().equals(realm.getPath()) {
    proxyState.getRow$realm().getIndex();
  }
}
```